From ee8d3df3292308097f4bfc46d6bb1931e66bb645 Mon Sep 17 00:00:00 2001
From: "Lay, Kuan Loon" <kuan.loon.lay@intel.com>
Date: Thu, 22 Sep 2016 11:36:44 +0800
Subject: [PATCH 12/14] IIO+Network: fix IIO gyroscope c code sample klocwork
 hit

Fix following klocwork hits:
- Line 373: void function returns value
- Line 517: Expression 'iio_config.trigger_name' used in the condition always
  yields the same result
- Line 523: Expression 'gyro_data.filter_data.buff' used in the condition always
  yields the same result
- Line 551: 'gyro_data.mqtt' might be used uninitialized in this function.
- Line 556: 'gyro_data.filter_data.buff' might be used uninitialized in this
  function

Upstream-Status: Pending

Signed-off-by: Lay, Kuan Loon <kuan.loon.lay@intel.com>
---
 .../iio-gyroscope-console-and-mqtt-publish.c       |   26 ++++++++------------
 1 file changed, 10 insertions(+), 16 deletions(-)

diff --git a/src/samples/iio+network/iio-gyroscope-console-and-mqtt-publish.c b/src/samples/iio+network/iio-gyroscope-console-and-mqtt-publish.c
index 2fc8bc6..9a33d63 100644
--- a/src/samples/iio+network/iio-gyroscope-console-and-mqtt-publish.c
+++ b/src/samples/iio+network/iio-gyroscope-console-and-mqtt-publish.c
@@ -370,7 +370,11 @@ iio_gyroscope_reader_cb(void *data, struct sol_iio_device *device)
     printf("%s\n", buffer);
 
 #ifdef MQTT
-    SOL_NULL_CHECK(gyro_data->mqtt, false);
+    if (gyro_data->mqtt == NULL)
+    {
+        SOL_WRN("mqtt is null");
+        return;
+    }
     if (sol_mqtt_get_connection_status(gyro_data->mqtt) == SOL_MQTT_CONNECTED) {
 
         payload = SOL_BUFFER_INIT_CONST(buffer, strlen(buffer));
@@ -404,6 +408,7 @@ main(int argc, char *argv[])
     int device_id;
 
     memset(&iio_config, 0, sizeof(struct sol_iio_config));
+    memset(&gyro_data, 0, sizeof(struct iio_gyroscope_data));
 #ifdef MQTT
     struct sol_mqtt_config mqtt_config = {
         SOL_SET_API_VERSION(.api_version = SOL_MQTT_CONFIG_API_VERSION,)
@@ -514,16 +519,12 @@ main(int argc, char *argv[])
 
     sol_run();
 
-    if (iio_config.trigger_name) {
-        free((char *)iio_config.trigger_name);
-        iio_config.trigger_name = NULL;
-    }
+    free((char *)iio_config.trigger_name);
+    iio_config.trigger_name = NULL;
 
 #ifdef DENOISE_MEDIAN
-    if (gyro_data.filter_data.buff) {
-        free(gyro_data.filter_data.buff);
-        gyro_data.filter_data.buff = NULL;
-    }
+    free(gyro_data.filter_data.buff);
+    gyro_data.filter_data.buff = NULL;
 #endif
 
     /* Must do sol_iio_close(device), it will disable IIO buffer.
@@ -547,19 +548,12 @@ error_iio:
          */
         sol_iio_close(device);
     }
-#ifdef MQTT
-    if (!gyro_data.mqtt)
-        SOL_WRN("Unable to create MQTT session");
-#endif
 
 #ifdef DENOISE_MEDIAN
     if (gyro_data.filter_data.buff) {
         free(gyro_data.filter_data.buff);
         gyro_data.filter_data.buff = NULL;
-    } else {
-        SOL_WRN("Unable to allocate denoise buffer");
     }
 #endif
-
     return -1;
 }
-- 
1.7.9.5

